errors/daemon.go

```diff
@@ -715,6 +715,15 @@ var (
 		HTTPStatusCode: http.StatusConflict,
 	})
 
+	// ErrorCodeContainerRestarting is generated when we try to top the
+	// container, but the container is restarting.
+	ErrorCodeContainerRestarting = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "CONTAINERRESTARTING",
+		Message:        "Container %s is restarting, wait until the container is running",
+		Description:    "An attempt was made to top the container, but the container is restarting",
```

Comment:
This is great, so I can also make use of this `Error` type now.
I'll copy you error message and do some rebase after this is merged. :+1: